sdk/data/azcosmos/cosmos_global_endpoint_manager.go

```diff
@@ -0,0 +1,170 @@
+package azcosmos
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"sync"
+	"time"
+
+	azlog "github.com/Azure/azure-sdk-for-go/sdk/azcore/log"
+	azruntime "github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+	"github.com/Azure/azure-sdk-for-go/sdk/internal/log"
+)
+
+const DefaultBackgroundRefreshInterval = 5 * time.Minute
+
+type globalEndpointManager struct {
+	client           *Client
+	preferredRegions []string
+	locationCache    *locationCache
+	refreshTicker    *time.Ticker
+	refreshDone      chan struct{}
+	sync.Mutex
+}
+
+// NewGlobalEndpointManager creates a new global endpoint manager.
+func NewGlobalEndpointManager(client *Client, preferredRegions []string) (*globalEndpointManager, error) {
```

Comment:
Same accessor comment for this. This is a constructor, but it's public, instead of internal:

```suggestion
func newGlobalEndpointManager(client *Client, preferredRegions []string) (*globalEndpointManager, error) {
```